### PR TITLE
Regression in release-4.0 branch

### PR DESCRIPTION
**Pull request by NicoNes**
This PR is my proposal to fix multiple regressions introduced by the initial commit (https://github.com/jakartaee/rest/commit/ae1f6ef7f4db9957079b8c383041a4537f33f35b) on branch "release-4-0" (see issue https://github.com/jakartaee/rest/issues/1165). 

My fix is made of 2 commits:
- First commit to rollback files where regression where spotted to their previous version from "master" branch.
- Second commit to apply back only relevant changes from  initial commit (https://github.com/jakartaee/rest/commit/ae1f6ef7f4db9957079b8c383041a4537f33f35b) on the files that I rollback in the first commit.

@spericas in your initial commit, you have massively replace all `Context` by `Entity` in javadoc in many classes (I listed them below). I do not revert these changes but I wonder if all of them are really relevant. If not antoher issue and PR will have to be opened to fix it.
- `jakarta.ws.rs.ext.Providers`
- `jakarta.ws.rs.core.UriInfo`
- `jakarta.ws.rs.core.SecurityContext`
- `jakarta.ws.rs.core.Configuration`
- `jakarta.ws.rs.core.HttpHeaders`

-- Nicolas

### CODE REVIEW

File: examples/src/main/java/jaxrs/examples/bootstrap/BasicJavaSeBootstrapExample.java [hunk 1]
```diff
@@ -46,21 +43,12 @@ private BasicJavaSeBootstrapExample() {
...
-            Runtime.getRuntime()
-                    .addShutdownHook(new Thread(() -> instance.stop()
-                            .thenAccept(stopResult -> System.out.printf("Stop result: %s [Native stop result: %s].%n",
-                                    stopResult, stopResult.unwrap(Object.class)))));
-
-            final Configuration actualConfigurarion = instance.configuration();
-            final URI uri = UriBuilder.newInstance().scheme(actualConfigurarion.protocol().toLowerCase())
-                    .host(actualConfigurarion.host()).port(actualConfigurarion.port())
-                    .path(actualConfigurarion.rootPath()).build();
+    public static void main(final String[] args) throws InterruptedException {
```
Comment (spericas) [inline examples/src/main/java/jaxrs/examples/bootstrap/BasicJavaSeBootstrapExample.java:46]: I honestly cannot recall why this change was made at that time.

File: jaxrs-api/src/main/java/jakarta/ws/rs/NotAuthorizedException.java [hunk 2]
```diff
@@ -16,16 +16,16 @@
...
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 
 import jakarta.ws.rs.core.Response;
 
-import static jakarta.ws.rs.core.HttpHeaders.WWW_AUTHENTICATE;
-import static jakarta.ws.rs.core.Response.Status.UNAUTHORIZED;
-
```
Comment (spericas) [inline jaxrs-api/src/main/java/jakarta/ws/rs/NotAuthorizedException.java:28]: This change is stylistic. Generally static imports are better at the end.

**Comment by spericas** [comment]
@NicoNes Thanks for looking into this. Generally looks good but let's drop changes that only update import orders, especially static ones. I believe the original 4.0 branch was created too early (before my large PR) and went stale.

**Comment by spericas** [comment]
@NicoNes Let's create another issue for the Context/Entity replacement problem you identified.

**Comment by NicoNes** [comment]
> @NicoNes Thanks for looking into this. Generally looks good but let's drop changes that only update import orders, especially static ones. I believe the original 4.0 branch was created too early (before my large PR) and went stale.

@spericas Ok I will drop those changes.  
I wasn't sure about what do to with those static import order changes because according to the GIT history, @mkarg commit (38772e3353f5f9159adfa008e4e25566df9ca6a5) about reorganizing the imports, moved these imports from down to top. So I believed your changes to move them back down again was not intentional.

**Review by spericas — APPROVED** [review]
LGTM

**Comment by spericas** [comment]
@mkarg Could you review this one too?

**Review by jansupol — APPROVED** [review]
